test/transportSpec.js

```diff
@@ -0,0 +1,382 @@
+        protocol: 'ipr'
+      })
+
+      this.params = {
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        secret: Buffer.from('shh_its_a_secret', 'base64'),
+      }
+
+      this.transfer = {
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        amount: '1',
+        to: 'test.example.alice.~ipr.GbLOVv3YyLo',
+        from: 'test.example.connie',
+        executionCondition: condition,
+        ilp: packet
+      }
+
+      // detect when autofulfill promise has resolved
+      this.fulfilled = new Promise((resolve) => {
+        this.callback = resolve
+      })
+
+      this.rejected = new Promise((resolve) => {
+        this.plugin.rejectIncomingTransfer = (id, msg) => {
+          resolve(msg)
+          return Promise.resolve()
+        }
+      })
+    })
+
+    it('should call fulfillCondition on a valid incoming transfer', function * () {
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+
+      // listener returns true for debug purposes
+      const res = yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      assert.isTrue(res[0])
+
+      yield this.fulfilled
+    })
+
+    it('should reject when it generates the wrong fulfillment', function * () {
+      this.transfer.executionCondition = 'garbage'
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+
+      // listener returns false for debug purposes
+      yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      yield this.rejected
+    })
+
+    it('should reject when packet details have been changed', function * () {
+      this.transfer.ilp = Buffer.concat([this.transfer.ilp, Buffer.from('00', 'hex')])
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+
+      // listener returns false for debug purposes
+      yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      yield this.rejected
+    })
+
+    it('should pass the fulfill function, transfer, decrypted data, destinationAmount, and destinationAccount to the callback', function * () {
+      const fulfilled = new Promise((resolve) => {
+        this.plugin.fulfillCondition = () => {
+          resolve()
+          return Promise.resolve()
+        }
+      })
+
+      this.callback = (details) => {
+        assert.isObject(details.transfer, 'must pass in transfer')
+        assert.isObject(details.data, 'must pass in decrypted data')
+        assert.isString(details.destinationAccount, 'must pass in account')
+        assert.isString(details.destinationAmount, 'must pass in amount')
+        assert.isFunction(details.fulfill, 'fulfill callback must be a function')
+        details.fulfill()
+      }
+
+      yield Transport.listen(this.plugin, this.params, this.callback, 'ipr')
+      yield this.plugin.emitAsync('incoming_prepare', this.transfer)
+      yield fulfilled
+    })
+
+    it('should reject if the listen callback rejects', function * () {
```

Comment:
This test description makes me think that if the callback returns a Promise that rejects that would also work. Is that true?